File: csharp/Lab3ForRefactoring/bnsit/patterns/Application.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Lab3ForRefactoring.bnsit.patterns.model;
using Lab3ForRefactoring.bnsit.patterns.logic.report;
using Lab3ForRefactoring.bnsit.patterns.logic;


namespace Lab3ForRefactoring.bnsit.patterns
{
    class Application
    {
        private const String PROMPT = "ees>";
        private const String HELLO_MESSAGE = "Welcome to Equipment Evidence System!";

        private ApplicationModel model = new ApplicationModel();

        private HelpSystemManager helpSystemManager = new HelpSystemManager();

        private Boolean running = false;

        public void Stop()
        {
            running = false;
        }

        public void Start()
        {
            running = true;

            Console.WriteLine(HELLO_MESSAGE);

            while (running)
            {
                Console.Write(PROMPT);
                String command = Console.ReadLine();

                if (command.Equals("hello"))
                {
                    Console.WriteLine(HELLO_MESSAGE);
                }
                else if (command.StartsWith("save"))
                {
                    String fileName = command.Split(' ')[1];
                    model.Save(fileName);
                }
                else if (command.StartsWith("load"))
                {
                    String fileName = command.Split(' ')[1];
                    model.Load(fileName);
                }
                else if (command.Equals("add_building"))
                {
                    addBuilidng();
                }
                else if (command.StartsWith("building_report"))
                {
                    String reportType = command.Split( ' ' )[1];

                    BuildingsReportGenerator generator = createGenerator( reportType );

                    Console.WriteLine( generator.Generate() );
                }
                else if (command.StartsWith("help")) {
                    string[] commandAndParams = command.Split( ' ' );

                    if (commandAndParams.Length == 1)
                    {
                        helpSystemManager.PrintAllHelp();
                    }
                    else
                    {
                        helpSystemManager.PrintHelp( commandAndParams[1] );
                    }
                }
                else if (command.Equals("exit"))
                {
                    Stop();
                }
            }
        }

        private BuildingsReportGenerator createGenerator(string reportType)
        {
            switch (reportType) {
                case "tree": return new TreeReportGenerator( model.Buildings );
                case "dashed": return new DahsedReportGenerator(model.Buildings);
                case "indented": return new IndentedReportGenerator( model.Buildings );
            }

            return null;
        }

        private void addBuilidng() {
            Console.Write("Name: "); ;
            string name = Console.ReadLine(); 
           
            Console.Write( "Address: " );
            string address = Console.ReadLine();

            BuildingBuilder builder = new BuildingBuilder();
            builder.BuildNewBuilding(name, address);

            Console.Write("First elevation No: ");
            int firstElevationNo = int.Parse(Console.ReadLine());

            Console.Write("Last elevation No: ");
            int lastElevationNo = int.Parse(Console.ReadLine());

            builder.AddEvevations(firstElevationNo, lastElevationNo);

            for (int i = firstElevationNo; i <= lastElevationNo; ++i)
            {
                Console.Write("How many room on elevation " + i + "?: ");
                int roomsNumber = int.Parse(Console.ReadLine());

                List<int> roomsNumbers = new List<int>();
                for (int j = 0; j < roomsNumber; ++j )
                {
                    Console.Write("Number of " + j + " room " + " of " + i + " elevation: ");
                    roomsNumbers.Add(int.Parse(Console.ReadLine()));
                }

                builder.AddRooms(i, roomsNumbers);
            }

            model.AddBuilding(builder.FinalBuilding);
        }

        static void Main(string[] args)
        {
            Application application = new Application();
            application.Start();
        }

    }
}

File: python/system-pomoc-do-refaktora/application.py
from model import ApplicationModel
from help_system import HelpSystemManager
from building_builder import BuildingBuilder
from reports import create_generator


class Application(object):
    PROMPT = "ees>"
    HELLO_MESSAGE = "Welcome to Equipment Evidence System."

    def __init__(self):
        self._running = False
        self._model = ApplicationModel()
        self._help_system = HelpSystemManager()

    def stop(self):
        self._running = False

    def run(self):
        self._running = True

        print(self.HELLO_MESSAGE)

        while self._running:
            command = input(self.PROMPT + " ")

            if command == "hello":
                print(self.HELLO_MESSAGE)

            elif command.startswith("save"):
                filename = command.split(" ")[1]
                self._model.save(filename)

            elif command.startswith("load"):
                filename = command.split(" ")[1]
                self._model.load(filename)

            elif command == "add_building":
                self._add_building()

            elif command.startswith("building_report"):
                report_type = command.split(" ")[1]
                generator = create_generator(report_type, self._model.buildings)
                print(generator.generate())

            elif command.startswith("help"):
                parts = command.split(" ")
                if len(parts) == 1:
                    self._help_system.print_all_help()
                else:
                    self._help_system.print_help(parts[1])

            elif command == "exit":
                self.stop()

    def _add_building(self):
        name = input("Name: ")
        address = input("Address: ")

        builder = BuildingBuilder()
        builder.build_new_building(name, address)

        first_floor_no = int(input("First elevation No: "))
        last_floor_no = int(input("Last elevation No: "))

        builder.add_floors(first_floor_no, last_floor_no)

        for i in range(first_floor_no, last_floor_no + 1):
            rooms_number = int(input("How many room on elevation " + str(i) + "?: "))
            room_numbers = []
            for j in range(rooms_number):
                room_no = int(input("Number of " + str(j) + " room of " + str(i) + " elevation: "))
                room_numbers.append(room_no)
            builder.add_rooms(i, room_numbers)

        self._model.add_building(builder.final_building)
